CMSIS/RTOS2/FreeRTOS/Source/cmsis_os2.c

```diff
@@ -2482,14 +2481,12 @@ osMemoryPoolId_t osMemoryPoolNew (uint32_t block_count, uint32_t block_size, con
     mp = NULL;
     sz = MEMPOOL_ARR_SIZE (block_count, block_size);
 
-    name = NULL;
     mem_cb = -1;
     mem_mp = -1;
 
     if (attr != NULL) {
-      if (attr->name != NULL) {
-        name = attr->name;
-      }
+      /* No need to check if NULL, NULL is OK */
```

Comment:
Currently this comment comes immediately after a NULL check and is deceiving. Add meaningful comment, something like:
`/* Take the name from attributes */`